### PR TITLE
Encrypt LDAP bind password in db with SECRET_KEY

### LINKED ISSUE #15460: Passwords for Database user and LDAP bind user are stored in plain text
- Gitea version (or commit ref): 1.13.7
- Git version: 2.14.0
- Operating system: Windows 10/Server 2012
  
  
  
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [x] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
- Log gist:

Bad security practice of storing passwords in plain text
## Description

While application does provide a warning that it stores password in plain text, I think we all agree that this is a big security issue independently from how many privileges user has.
There is an existing module modules/secret/secret.go that allows for two way encryption. Why is it not used to encrypt password values stored in app.ini and in the database in case of LDAP auth.
...

### PR DESCRIPTION
**Pull request by zeripath**
The LDAP source bind password are currently stored in plaintext in the db
This PR simply encrypts them with the setting.SECRET_KEY.

Fix #15460

## :warning: WARNING :warning:

Changing the `SECRET_KEY` in the app.ini will now cause these passwords to become unreadable.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by techknowlogick** [comment]
I think this would be a great use case for @lafriks secrets PR

**Review by lunny — APPROVED** [review]

**Review by noerw — APPROVED** [review]

**Comment by justusbunsi** [comment]
Does this PR mean that the hint (... stored in plaintext) in the LDAP configuration is not necessary anymore?

File: docs/content/doc/features/authentication.en-us.md [hunk 1]
```diff
@@ -88,8 +88,8 @@ Adds the following fields:
 - Bind Password (optional)
 
   - The password for the Bind DN specified above, if any. _Note: The password
-    is stored in plaintext at the server. As such, ensure that the Bind DN
-    has as few privileges as possible._
+    is stored in encrypted with the SECRET_KEY on the server. It is still recommended
```
Comment (justusbunsi) [inline docs/content/doc/features/authentication.en-us.md]: _is stored ~in~ encrypted with the_. Although it seems to work with/without the "in". 🤔

**Comment by silverwind** [comment]
Removed the mentioned UI warning in the LDAP settings in https://github.com/go-gitea/gitea/pull/15547/commits/987f19290f06cc12680014c326175115df838fa2.

**Comment by zeripath** [comment]
make lgtm work

**Comment by silverwind** [comment]
BTW do we need to migrate existing entries?

**Comment by zeripath** [comment]
The data will be encrypted on saving.

**Comment by silverwind** [comment]
Ah, I see its using a conditional decryption based on that `BindPasswordEncrypt` variable, so I guess it's fine actually.

**Pull request merged**